app/models/piece.rb

```diff
@@ -152,3 +152,36 @@ def in_self_check?
     end
   end
 end
+
+def valid_castle_black?(position_x,position_y)
+  king = pieces.find_by(type: 'King', color:'black')
+  rook = pieces.find_by(type: 'rook', color:'black')
+  !in_check?
+  king.move_count && rook.move_count == 0
+  if king.valid_move?(1,7)
+    king.move_to(1,7)
```

Comment:
there seems to be a little disconnect between  what the method name suggest what it does (`valid_castle_black?`) and what it actually does - actually performing the castling.. as a convention, we try to name methods with `?` which will return true or false without any side effects.. 
 
maybe we could split it into 2 methods, one checking if its allowed and the other actually executing the move..

so that the overall thing would be sth like:

```
def castle_black!
   make_castle_move_for_black!  if valid_castle_black? 
end

def valid_castle_black?
  # returns true or false
end

def make_castle_move_for_black!
# make sthe actual castle move
end 
  ```